repair/repair.cc

```diff
@@ -279,6 +280,28 @@ repair_status tracker::get(int id) {
     }
 }
 
+future<repair_status> tracker::get_status_until_finished(int id, std::chrono::steady_clock::time_point timeout) {
+    if (id >= _next_repair_command) {
+        return make_exception_future<repair_status>(std::runtime_error(format("unknown repair id {:d}", id)));
+    }
+    return repeat_until_value([this, id, timeout] {
+        auto it = _status.find(id);
+        if (it == _status.end()) {
+            return make_ready_future<std::optional<repair_status>>(repair_status::SUCCESSFUL);
```

Comment:
I don't like it much, but ok.

Reply:
If a node restarts, then a failed repair can become successful, no? We have to make repair results persistent.

Reply:
No. When a node restarts,  the next repair id is 1. Status queries of exiting repair repair id (which >=1) will fail with "unknown repair id " because repair knows no repair job is started yet.  

After we transit to uuid repair ids, we can store them in a system table.

Reply:
Ah. It's still vulnerable to to collisions, but that's the best we can do for now.